test/cypress/integration/simplified-account/service-settings/worldpay-details/worldpay-details.cy.js

```diff
@@ -0,0 +1,183 @@
+const userStubs = require('@test/cypress/stubs/user-stubs')
+const gatewayAccountStubs = require('@test/cypress/stubs/gateway-account-stubs')
+const ROLES = require('@test/fixtures/roles.fixtures')
+const { WORLDPAY, STRIPE } = require('@models/payment-providers')
+
+const USER_EXTERNAL_ID = 'user-123-abc'
+const SERVICE_EXTERNAL_ID = 'service-456-def'
+const GATEWAY_ACCOUNT_ID = 11
+const ACCOUNT_TYPE = 'test'
+
+const VALID_MOTO_MERCHANT_CODE = 'AVALIDMERCHANTCODEMOTO'
+const VALID_WORLDPAY_USERNAME = 'worldpay-user'
+
+const setupStubs = (opts = {}) => {
+  const options = Object.assign({}, {
+    role: 'admin',
+    paymentProvider: WORLDPAY,
+    credentials: {}
+  }, opts)
+
+  cy.task('setupStubs', [
+    userStubs.getUserSuccess({
+      userExternalId: USER_EXTERNAL_ID,
+      gatewayAccountId: GATEWAY_ACCOUNT_ID,
+      serviceName: { en: 'My cool service' },
+      serviceExternalId: SERVICE_EXTERNAL_ID,
+      role: ROLES[options.role],
+      features: 'degatewayaccountification' // TODO remove features once simplified accounts are live
+    }),
+    gatewayAccountStubs.getAccountByServiceIdAndAccountType(SERVICE_EXTERNAL_ID, ACCOUNT_TYPE, {
+      gateway_account_id: GATEWAY_ACCOUNT_ID,
+      payment_provider: options.paymentProvider,
+      gateway_account_credentials: [{
+        payment_provider: options.paymentProvider,
+        credentials: options.credentials
+      }],
+      allow_moto: true
+    })
+  ])
+}
+
+describe('Worldpay details settings', () => {
+  beforeEach(() => {
+    cy.setEncryptedCookies(USER_EXTERNAL_ID)
+  })
+
+  describe('Worldpay details landing page', () => {
+    describe('for an admin user', () => {
+      describe('when some tasks are incomplete', () => {
+        beforeEach(() => {
+          setupStubs()
+        })
+
+        it('should show the correct heading and title', () => {
+          cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/worldpay-details`)
```

Comment:
These visits could go into line 52 after the `setupStubs()` call.

Reply:
imo visiting the page constitutes part of the test rather than the setup, so my preference is to keep the `cy.visit` calls in the test. I think it also helps make it clear exactly where the flow that is being tested begins